core/src/main/scala/scalaz/NonEmptyList.scala

```diff
@@ -42,10 +44,20 @@ final class NonEmptyList[+A] private[scalaz](val head: A, val tail: List[A]) {
     nel(bb.head, bb.tail)
   }
 
-  def distinct[AA >: A](implicit A: Order[AA]): NonEmptyList[A] =
-    (list.distinct: @unchecked) match {
-      case x :: xs => nel(x, xs)
-    }
+  def distinct[AA>:A](implicit A: Order[AA]): NonEmptyList[AA] = {
+    @tailrec def loop(src: List[A], seen: ISet[AA], acc: NonEmptyList[AA]): NonEmptyList[AA] =
+      src match {
+        case h :: t =>
+          if(seen.notMember(h)){
```

Comment:
really minor, but spaces before and after braces. We should incorporate an autoformatter…